### PR TITLE
[9.x] Add analytics for meilisearch engine

### PR DESCRIPTION
**Pull request by mmachatschek**
This adds user agent information to the meilisearch client to support anonymous telemetry on Meilisearch side  

https://github.com/laravel/scout/pull/678#issuecomment-1359461649

### CODE REVIEW

**Review by driesvints — APPROVED** [review]
Thanks @mmachatschek!

File: src/ScoutServiceProvider.php [hunk 1]
```diff
@@ -22,11 +23,21 @@ public function register()
     {
         $this->mergeConfigFrom(__DIR__.'/../config/scout.php', 'scout');
 
-        if (class_exists(MeiliSearch::class)) {
-            $this->app->singleton(MeiliSearch::class, function ($app) {
+        if (class_exists(MeilisearchClient::class)) {
```
Comment (brunoocasali) [inline src/ScoutServiceProvider.php:26]: @mmachatschek, in this case, you are counting on the new meilisearch-php version (after the module case changes you did).

Should we add a `composer.json` protection? Or change the classes `MeilisearchClient` to the old ones?

Comment (brunoocasali) [reply src/ScoutServiceProvider.php:26]: Or are my concerns not applicable here due to the PHP case-insensitiveness?

Comment (mmachatschek) [reply src/ScoutServiceProvider.php]: Thanks @brunoocasali the case insensitivity wouldn't have complained about that but for consistency I updated the code 👍

**Comment by driesvints** [comment]
@mmachatschek I merged this PR into 10.x, can you update your PR?

**Comment by d8vjork** [comment]
Somehow ScoutServiceProvider is now failing on my company's CI when installing composer packages (at dumpautoload):

image

image

We can't ignore autoload, so we downgrade Scout version to one before this PR anyway I think this must be fixed

**Comment by d8vjork** [comment]
Ok, issue found with latest composer release: 2.5.1

Solved by downgrading to 2.4.x, seems related to Composer

**Comment by driesvints** [comment]
Please see https://github.com/laravel/scout/pull/687 as this is a change in meilisearch php

**Comment by d8vjork** [comment]
> Please see #687 as this is a change in meilisearch php

I see, seems thought downgrading Scout solved it from now

**Pull request merged**